File: PE3.py
###Correct code
'''output
Highest prime factor is  6857

real	0m0.030s
user	0m0.028s
sys	0m0.000s'''
#Better Code in short

n=600851475143

i = 2
while i * i < n:
    while n % i == 0:
        n = n / i
    i = i + 1
print "Highest prime factor is ",n

File: PE3.c
/**********************************************************************
*This Code wont work for higher numbers								  *
*																	  *
***********************************************************************/

#include<stdio.h>

void main(){
	long unsigned int i=2,n,m;
	printf("enter the number");
	scanf("%d",&n);
	m=n;
	while(i*i<n){
	while(n%i==0){
		n/=i;
	}
		i+=1;
	}
	printf("Highest prime factor of %ul is %ul\n",m,n);
}
